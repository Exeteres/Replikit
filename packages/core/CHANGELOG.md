# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.17.2](https://github.com/Exeteres/Replikit/compare/v0.17.1...v0.17.2) (2020-08-24)

**Note:** Version bump only for package @replikit/core





## [0.17.1](https://github.com/Exeteres/Replikit/compare/v0.17.0...v0.17.1) (2020-08-21)

**Note:** Version bump only for package @replikit/core





# [0.17.0](https://github.com/Exeteres/Replikit/compare/v0.16.0...v0.17.0) (2020-08-21)


### Features

* **core:** Add Identifier type ([e97ef90](https://github.com/Exeteres/Replikit/commit/e97ef90d24068fe4c16842b17bae0b8e6570ec47))





# [0.16.0](https://github.com/Exeteres/Replikit/compare/v0.15.1...v0.16.0) (2020-08-18)


### Bug Fixes

* **core:** Fix bug in text tokenization system ([43404bf](https://github.com/Exeteres/Replikit/commit/43404bffdb878e1f4abd18f69970cd6520251dd9))


### Features

* **core:** Add support for message headers ([5754c94](https://github.com/Exeteres/Replikit/commit/5754c94e0f89edb4ff3dde4fe410ade303bfb58f))





## [0.15.1](https://github.com/Exeteres/Replikit/compare/v0.15.0...v0.15.1) (2020-08-13)

**Note:** Version bump only for package @replikit/core





# [0.15.0](https://github.com/Exeteres/Replikit/compare/v0.14.0...v0.15.0) (2020-08-13)

**Note:** Version bump only for package @replikit/core





# [0.14.0](https://github.com/Exeteres/Replikit/compare/v0.13.0...v0.14.0) (2020-08-05)


### Bug Fixes

* **cli:** Fix typings ([9851fe5](https://github.com/Exeteres/Replikit/commit/9851fe5b87e07318da5c085f4b1bbddb23984136))





# [0.13.0](https://github.com/Exeteres/Replikit/compare/v0.12.1...v0.13.0) (2020-08-04)


### Bug Fixes

* **core:** Fix constructor type ([9687c48](https://github.com/Exeteres/Replikit/commit/9687c48773917919a2d8c38add27fba15cc11f00))


### Features

* **core:** Add ability to use pipe method in builders ([95c3cca](https://github.com/Exeteres/Replikit/commit/95c3cca0d1f4c5203fde3386c3775ecc77b92a1b))





# [0.12.0](https://github.com/Exeteres/Replikit/compare/v0.11.2...v0.12.0) (2020-07-30)


### Bug Fixes

* **core:** Rename query field to text ([4b0a930](https://github.com/Exeteres/Replikit/commit/4b0a930a4a840efcf56bdc6dfcf0b77a9430a30c))


### Features

* **core:** Refactor types and add support for inline queries ([2e52258](https://github.com/Exeteres/Replikit/commit/2e52258587cf0f2d31061c991eb2489ec7acda9e))





## [0.11.2](https://github.com/Exeteres/Replikit/compare/v0.11.1...v0.11.2) (2020-07-24)

**Note:** Version bump only for package @replikit/core





## [0.11.1](https://github.com/Exeteres/Replikit/compare/v0.11.0...v0.11.1) (2020-07-24)

**Note:** Version bump only for package @replikit/core






# [0.11.0](https://github.com/Exeteres/Replikit/compare/v0.10.1...v0.11.0) (2020-07-24)


### Bug Fixes

* **core:** Fix deepmerge ([132b184](https://github.com/Exeteres/Replikit/commit/132b18493e02603c195af70a2cfff0e9c31dc383))


### Features

* **sessions:** Add sessions package ([e7a0adf](https://github.com/Exeteres/Replikit/commit/e7a0adf3bbe35091bad6dcf28395e3c389a20a83))





## [0.10.1](https://github.com/Exeteres/Replikit/compare/v0.10.0...v0.10.1) (2020-06-30)

**Note:** Version bump only for package @replikit/core






# [0.10.0](https://github.com/Exeteres/Replikit/compare/v0.9.0...v0.10.0) (2020-06-29)

**Note:** Version bump only for package @replikit/core





# [0.9.0](https://github.com/Exeteres/Replikit/compare/v0.8.0...v0.9.0) (2020-06-28)


### Features

* Add advanced metadata processing ([54c7820](https://github.com/Exeteres/Replikit/commit/54c782021f5ebc16d784f210d273f8d7490d2af5))





# [0.8.0](https://github.com/Exeteres/Replikit/compare/v0.7.2...v0.8.0) (2020-06-27)


### Features

* **core:** Add core support for replies ([21f3318](https://github.com/Exeteres/Replikit/commit/21f3318f7bd031233856e7a1e1ddd29d73ef5f04))





## [0.7.2](https://github.com/Exeteres/Replikit/compare/v0.7.1...v0.7.2) (2020-06-23)

**Note:** Version bump only for package @replikit/core





# [0.7.0](https://github.com/Exeteres/Replikit/compare/v0.6.0...v0.7.0) (2020-06-23)

**Note:** Version bump only for package @replikit/core





# [0.6.0](https://github.com/Exeteres/Replikit/compare/v0.5.3...v0.6.0) (2020-06-21)

**Note:** Version bump only for package @replikit/core





## [0.5.3](https://github.com/Exeteres/Replikit/compare/v0.5.2...v0.5.3) (2020-06-20)

**Note:** Version bump only for package @replikit/core





## [0.5.2](https://github.com/Exeteres/Replikit/compare/v0.5.1...v0.5.2) (2020-06-20)

**Note:** Version bump only for package @replikit/core





# 0.5.0 (2020-06-20)


### Features

* **cli:** Add build command ([5c9dc57](https://github.com/Exeteres/Replikit/commit/5c9dc57b9ac2428f4970096dcfc6ae75b2e4dd27))






# [0.4.0](https://github.com/Exeteres/Replikit/compare/v0.3.0...v0.4.0) (2020-06-19)


### Features

* **cli:** Add build command ([0212d7f](https://github.com/Exeteres/Replikit/commit/0212d7f9c87d517b8cf220be9b32376551c3603b))





# [0.3.0](https://github.com/Exeteres/Replikit/compare/v0.2.0...v0.3.0) (2020-06-14)

**Note:** Version bump only for package @replikit/core





# [0.2.0](https://github.com/Exeteres/Replikit/compare/v0.1.0...v0.2.0) (2020-06-13)

**Note:** Version bump only for package @replikit/core
