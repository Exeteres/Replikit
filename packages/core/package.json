{
    "name": "@replikit/core",
    "version": "0.11.1",
    "license": "MIT",
    "main": "dist/index.js",
    "typings": "dist/index.d.ts",
    "files": [
        "dist",
        "typings"
    ],
    "scripts": {
        "build": "tsc -p tsconfig.build.json"
    }
}
