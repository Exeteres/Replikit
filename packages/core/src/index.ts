/// <reference path="../typings/index.d.ts" />

export * from "./models/channelType";
export * from "./models/attachmentType";
export * from "./models/textTokenKind";
export * from "./models/textTokenProp";
export * from "./models/logLevel";

export * from "./composition/compositionItem";
export * from "./composition/compositionFactory";
export * from "./composition/compositionContext";
export * from "./composition/composition";

export * from "./errors";
export * from "./utils";
export * from "./hookStorage";
export * from "./config";
export * from "./eventHandler";
export * from "./cacheManager";
export * from "./textTokenizer";
export * from "./textFormatter";
export * from "./controller";
export * from "./controllerStorage";

export * from "./logging/consoleLogProvider";
export * from "./logging/logManager";
export * from "./logging/logger";

import "./startup";
