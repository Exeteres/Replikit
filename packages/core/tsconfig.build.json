{
    "extends": "../../tsconfig.build.json",
    "compilerOptions": {
        "outDir": "dist",
        "paths": {
            "@replikit/core": ["core/src"]
        }
    },
    "include": ["src"]
}
