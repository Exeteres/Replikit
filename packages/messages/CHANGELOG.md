# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.22.1](https://github.com/Exeteres/Replikit/compare/v0.22.0...v0.22.1) (2020-10-23)

**Note:** Version bump only for package @replikit/messages





# [0.22.0](https://github.com/Exeteres/Replikit/compare/v0.21.0...v0.22.0) (2020-10-23)

**Note:** Version bump only for package @replikit/messages





# [0.21.0](https://github.com/Exeteres/Replikit/compare/v0.20.0...v0.21.0) (2020-10-17)

**Note:** Version bump only for package @replikit/messages





# [0.20.0](https://github.com/Exeteres/Replikit/compare/v0.19.0...v0.20.0) (2020-09-28)


### Features

* Add ability to create multiple button rows ([2a83a74](https://github.com/Exeteres/Replikit/commit/2a83a7432c102ff77e71cd4d542fdb4ec99ac7ad))





# [0.19.0](https://github.com/Exeteres/Replikit/compare/v0.18.2...v0.19.0) (2020-09-18)


### Features

* **messages:** Add addButton method ([1d7f3a0](https://github.com/Exeteres/Replikit/commit/1d7f3a0adb2b31e02707f4093e0ec7d828797403))
* **messages:** Add OutMessageLike type and utils ([a9fb59c](https://github.com/Exeteres/Replikit/commit/a9fb59cb432419fadd3a51e79d135ac29c9087b5))
* **messages:** Add token factories and support for buttons ([2275483](https://github.com/Exeteres/Replikit/commit/227548389a70cf64756c298951241ea96fbd77c4))





## [0.18.2](https://github.com/Exeteres/Replikit/compare/v0.18.1...v0.18.2) (2020-09-09)

**Note:** Version bump only for package @replikit/messages





## [0.18.1](https://github.com/Exeteres/Replikit/compare/v0.18.0...v0.18.1) (2020-09-09)

**Note:** Version bump only for package @replikit/messages





# [0.18.0](https://github.com/Exeteres/Replikit/compare/v0.17.3...v0.18.0) (2020-09-09)

**Note:** Version bump only for package @replikit/messages





## [0.17.3](https://github.com/Exeteres/Replikit/compare/v0.17.2...v0.17.3) (2020-08-25)

**Note:** Version bump only for package @replikit/messages





## [0.17.2](https://github.com/Exeteres/Replikit/compare/v0.17.1...v0.17.2) (2020-08-24)

**Note:** Version bump only for package @replikit/messages





## [0.17.1](https://github.com/Exeteres/Replikit/compare/v0.17.0...v0.17.1) (2020-08-21)

**Note:** Version bump only for package @replikit/messages





# [0.17.0](https://github.com/Exeteres/Replikit/compare/v0.16.0...v0.17.0) (2020-08-21)

**Note:** Version bump only for package @replikit/messages





# [0.16.0](https://github.com/Exeteres/Replikit/compare/v0.15.1...v0.16.0) (2020-08-18)


### Features

* **messages:** Add methods to work with message headers ([f412a28](https://github.com/Exeteres/Replikit/commit/f412a280c4a185a8066fec2b8492bc4afd8d8431))





## [0.15.1](https://github.com/Exeteres/Replikit/compare/v0.15.0...v0.15.1) (2020-08-13)

**Note:** Version bump only for package @replikit/messages





# [0.15.0](https://github.com/Exeteres/Replikit/compare/v0.14.0...v0.15.0) (2020-08-13)

**Note:** Version bump only for package @replikit/messages





# [0.14.0](https://github.com/Exeteres/Replikit/compare/v0.13.0...v0.14.0) (2020-08-05)


### Features

* **messages:** Add ability to use addLine method without text ([3886f11](https://github.com/Exeteres/Replikit/commit/3886f113aa94943afea2bd5f4f94462eccec055e))





# [0.13.0](https://github.com/Exeteres/Replikit/compare/v0.12.1...v0.13.0) (2020-08-04)


### Features

* **messages:** Add ability to use pipe method in MessageBuilder ([f5fe039](https://github.com/Exeteres/Replikit/commit/f5fe0391708753b7404e6c695a77cd793e83120c))
* **messages:** Add alias to use message with metadata as metadata ([e315a25](https://github.com/Exeteres/Replikit/commit/e315a256dfeef85137f640e08176d14c84a1638f))





# [0.12.0](https://github.com/Exeteres/Replikit/compare/v0.11.2...v0.12.0) (2020-07-30)

**Note:** Version bump only for package @replikit/messages





## [0.11.2](https://github.com/Exeteres/Replikit/compare/v0.11.1...v0.11.2) (2020-07-24)

**Note:** Version bump only for package @replikit/messages





## [0.11.1](https://github.com/Exeteres/Replikit/compare/v0.11.0...v0.11.1) (2020-07-24)

**Note:** Version bump only for package @replikit/messages






# [0.11.0](https://github.com/Exeteres/Replikit/compare/v0.10.1...v0.11.0) (2020-07-24)


### Features

* **messages:** Add MessageLike type ([046ca1b](https://github.com/Exeteres/Replikit/commit/046ca1b8a19308790f3e0a424a0cb9f3cb7dc7dd))





## [0.10.1](https://github.com/Exeteres/Replikit/compare/v0.10.0...v0.10.1) (2020-06-30)

**Note:** Version bump only for package @replikit/messages






# [0.10.0](https://github.com/Exeteres/Replikit/compare/v0.9.0...v0.10.0) (2020-06-29)

**Note:** Version bump only for package @replikit/messages





# [0.9.0](https://github.com/Exeteres/Replikit/compare/v0.8.0...v0.9.0) (2020-06-28)


### Features

* Add advanced metadata processing ([54c7820](https://github.com/Exeteres/Replikit/commit/54c782021f5ebc16d784f210d273f8d7490d2af5))





# [0.8.0](https://github.com/Exeteres/Replikit/compare/v0.7.2...v0.8.0) (2020-06-27)


### Features

* **messages:** Add MessageBuilder methods for working with replies ([015e3f2](https://github.com/Exeteres/Replikit/commit/015e3f23b3f166b46aecb2b98f6f447661b8576d))





## [0.7.2](https://github.com/Exeteres/Replikit/compare/v0.7.1...v0.7.2) (2020-06-23)

**Note:** Version bump only for package @replikit/messages





# [0.7.0](https://github.com/Exeteres/Replikit/compare/v0.6.0...v0.7.0) (2020-06-23)

**Note:** Version bump only for package @replikit/messages





# [0.6.0](https://github.com/Exeteres/Replikit/compare/v0.5.3...v0.6.0) (2020-06-21)

**Note:** Version bump only for package @replikit/messages





## [0.5.3](https://github.com/Exeteres/Replikit/compare/v0.5.2...v0.5.3) (2020-06-20)

**Note:** Version bump only for package @replikit/messages





## [0.5.2](https://github.com/Exeteres/Replikit/compare/v0.5.1...v0.5.2) (2020-06-20)

**Note:** Version bump only for package @replikit/messages





# 0.5.0 (2020-06-20)

**Note:** Version bump only for package @replikit/messages






# [0.4.0](https://github.com/Exeteres/Replikit/compare/v0.3.0...v0.4.0) (2020-06-19)

**Note:** Version bump only for package @replikit/messages





# [0.3.0](https://github.com/Exeteres/Replikit/compare/v0.2.0...v0.3.0) (2020-06-14)

**Note:** Version bump only for package @replikit/messages





# [0.2.0](https://github.com/Exeteres/Replikit/compare/v0.1.0...v0.2.0) (2020-06-13)

**Note:** Version bump only for package @replikit/messages
