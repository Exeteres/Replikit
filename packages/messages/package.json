{
    "name": "@replikit/messages",
    "version": "0.22.1",
    "license": "MIT",
    "main": "dist/index.js",
    "typings": "dist/index.d.ts",
    "files": [
        "dist",
        "typings"
    ],
    "scripts": {
        "build": "tsc -p tsconfig.build.json"
    },
    "dependencies": {
        "@replikit/core": "^0.22.1"
    }
}
