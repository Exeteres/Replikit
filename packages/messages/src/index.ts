export * from "./shortcuts";
export * from "./utils";
export * from "./messageBuilder";
