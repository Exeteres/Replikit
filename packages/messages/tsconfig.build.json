{
    "extends": "../../tsconfig.build.json",
    "compilerOptions": {
        "outDir": "dist",
        "paths": {
            "@replikit/messages": ["messages/src"]
        }
    },
    "include": ["src"]
}
