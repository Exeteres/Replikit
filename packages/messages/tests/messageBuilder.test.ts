import { MessageBuilder } from "@replikit/messages";
import { AttachmentType, TextTokenKind } from "@replikit/core";

describe("MessageBuilder", () => {
    it("should create a message with text", () => {
        const message = new MessageBuilder()
            .addText("firstLine - ")
            .addLine("firstLine")
            .addText("secondLine")
            .build();
        expect(message).toMatchSnapshot();
    });

    it("should create a message with code", () => {
        const message = new MessageBuilder().addCode("code").build();
        expect(message).toMatchSnapshot();
    });

    it("should create a message with an attachment", () => {
        const message = new MessageBuilder()
            .addAttachment({ id: "test", type: AttachmentType.Photo })
            .build();
        expect(message).toMatchSnapshot();
    });

    it("should create a message with an array of attachments", () => {
        const message = new MessageBuilder()
            .addAttachments([{ id: "test", type: AttachmentType.Photo }])
            .build();
        expect(message).toMatchSnapshot();
    });

    it("should create a message with an attachment by url", () => {
        const message = new MessageBuilder()
            .addAttachmentByUrl(AttachmentType.Photo, "https://path/to/attachment.png")
            .build();
        expect(message).toMatchSnapshot();
    });

    it("should create a message with reply", () => {
        const message = new MessageBuilder().addReply({ messageIds: [123] }).build();
        expect(message).toMatchSnapshot();
    });

    it("should create a message with an array of tokens", () => {
        const message = new MessageBuilder()
            .addTokens([{ kind: TextTokenKind.Text, text: "test", props: [] }])
            .build();
        expect(message).toMatchSnapshot();
    });
});
