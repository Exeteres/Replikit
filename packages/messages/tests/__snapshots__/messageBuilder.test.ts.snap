// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBuilder should create a message with an array of attachments 1`] = `
Object {
  "attachments": Array [
    Object {
      "id": "test",
      "type": 0,
    },
  ],
  "forwarded": Array [],
  "tokens": Array [],
}
`;

exports[`MessageBuilder should create a message with an array of tokens 1`] = `
Object {
  "attachments": Array [],
  "forwarded": Array [],
  "tokens": Array [
    Object {
      "kind": 0,
      "props": Array [],
      "text": "test",
    },
  ],
}
`;

exports[`MessageBuilder should create a message with an attachment 1`] = `
Object {
  "attachments": Array [
    Object {
      "id": "test",
      "type": 0,
    },
  ],
  "forwarded": Array [],
  "tokens": Array [],
}
`;

exports[`MessageBuilder should create a message with an attachment by url 1`] = `
Object {
  "attachments": Array [
    Object {
      "id": "c103c4603fe632d0f4119b0362b5b1a48d2db6d5",
      "type": 0,
      "url": "https://path/to/attachment.png",
    },
  ],
  "forwarded": Array [],
  "tokens": Array [],
}
`;

exports[`MessageBuilder should create a message with code 1`] = `
Object {
  "attachments": Array [],
  "forwarded": Array [],
  "tokens": Array [
    Object {
      "kind": 0,
      "props": Array [
        5,
      ],
      "text": "code",
    },
  ],
}
`;

exports[`MessageBuilder should create a message with empty code line 1`] = `
Object {
  "attachments": Array [],
  "forwarded": Array [],
  "tokens": Array [
    Object {
      "kind": 0,
      "props": Array [
        5,
      ],
      "text": "
",
    },
  ],
}
`;

exports[`MessageBuilder should create a message with empty line 1`] = `
Object {
  "attachments": Array [],
  "forwarded": Array [],
  "tokens": Array [
    Object {
      "kind": 0,
      "props": Array [],
      "text": "
",
    },
  ],
}
`;

exports[`MessageBuilder should create a message with reply 1`] = `
Object {
  "attachments": Array [],
  "forwarded": Array [],
  "reply": Object {
    "messageIds": Array [
      123,
    ],
  },
  "tokens": Array [],
}
`;

exports[`MessageBuilder should create a message with text 1`] = `
Object {
  "attachments": Array [],
  "forwarded": Array [],
  "tokens": Array [
    Object {
      "kind": 0,
      "props": Array [],
      "text": "firstLine - ",
    },
    Object {
      "kind": 0,
      "props": Array [],
      "text": "firstLine
",
    },
    Object {
      "kind": 0,
      "props": Array [],
      "text": "secondLine",
    },
  ],
}
`;
